### ISSUE TITLE
"Latest Version" on browse section

### ISSUE THREAD

**Issue by MultipedBeatle** [modrinth/knossos#456]
### Is your feature request related to a problem? Please describe.

I have Mod Manager installed, which tells me when a mod can be updated. It is not always accurate, and its gotten tedious to go to the website, search the mod, make sure its the version for the modloader, then check the latest version with my MultiMC loader list. Not to mention that sometimes the sidebar in the mod's page has not updated properly.

### Describe the solution you'd like

Having the latest version of the mod, as uploaded by the author, displayed on the list of mods page.

### Describe alternatives you've considered

I'm fine with clicking into the mod to view the version, but having it there at a glance when doing a mass update would be nice. There isn't really an alternative besides clicking in at the moment.

### Additional context

![image](https://user-images.githubusercontent.com/55055851/163686290-202f69da-9bf1-4c47-9ce1-2d0c7801709e.png)
I took a screenshot of a Sodium to use as my example, as its the first mod listed on modrinth.com/mods. The text I placed were some example areas of where they could go, but I believe having the latest version placed anywhere would be fine.

**Comment by wafflecoffee** [comment]
The issue here is that "latest" is a bit of a misnomer. There are several mods that support multiple versions and loaders, and so displaying `1.2.3+forge-1.17` as the latest when there are different loaders and MC versions is counterintuitive. Additionally, the search API doesn't provide this sort of information at all, though we could add it if we really wanted to (even though we really shouldn't).

Really, the best solution here would be implemented in Mod Manager itself, such as perhaps giving confirmation that you want to update to a given version or to give a list of different versions. However, that's not something for us to do considering this is an external application.
In fact, looking through the source code, Mod Manager's way of finding what version of a mod you already have is *really* convoluted. @DeathsGun Would it be possible for Mod Manager to use [this route](https://docs.modrinth.com/api-spec/#operation/getLatestVersionFromHash) for determining whether a mod should be updated? You might need to update everything else to use v2 as well, the migration guide for which can be found [here](https://docs.modrinth.com/docs/migrations/v1-to-v2/).

**Comment by DeathsGun** [comment]
> In fact, looking through the source code, Mod Manager's way of finding what version of a mod you already have is *really* convoluted. @DeathsGun Would it be possible for Mod Manager to use [this route](https://docs.modrinth.com/api-spec/#operation/getLatestVersionFromHash) for determining whether a mod should be updated? You might need to update everything else to use v2 as well, the migration guide for which can be found [here](https://docs.modrinth.com/docs/migrations/v1-to-v2/).

Yeah ModManager's process for finding updates is currently a bit inefficient. I'm currently working on a rewrite using the route you have provided which is version independent from Minecraft and ModMenu.
